runtime/integration-test/src/xcm/kusama_xcm_transfer.rs

```diff
@@ -193,6 +191,139 @@ fn transfer_asset() {
 	});
 }
 
+#[test]
+fn receive_asset() {
+	TestNet::reset();
+	let unit = unit(18);
+	let registered_asset_id = RELAY_ASSET_ID + 1;
+
+	// transfer native sibling asset
+	Sibling::execute_with(|| {
+		assert_ok!(XTokens::transfer(
+			RuntimeOrigin::signed(ALICE.into()),
+			NATIVE_ASSET_ID,
+			20 * unit,
+			Box::new(
+				MultiLocation::new(
+					1,
+					X2(
+						Parachain(MANGATA_ID),
+						Junction::AccountId32 { network: NetworkId::Any, id: BOB.into() }
+					)
+				)
+				.into()
+			),
+			WeightLimit::Limited(600_000_000),
+		));
+
+		assert_eq!(Tokens::free_balance(NATIVE_ASSET_ID, &AccountId::from(ALICE)), 80 * unit);
+	});
+
+	// no asset found -> no FPS -> too expensive
+	Mangata::execute_with(|| {
```

Comment:
is it dependant on `Sibling::execute_with` ? Can we maybe split it into multiple smaller test cases ?